### ISSUE TITLE
CMake fault with FMS 2024.02 on AWS Unbuntu 22.04 EC2 instance

### ISSUE THREAD

**Issue by srherbener** [JCSDA/spack-stack#1269]
**Describe the bug**

After successfully building spack-stack on an Ubuntu 22.04 EC2 instance, I ran into a cmake issue when running ecbuild on jedi-bundle, that looked like this:
```
-- Configuring done (148.0s)
CMake Error at /home/ubuntu/spack-stack/manual-testing/spst-boost-gcc4intel/envs/ue-intel-2021.10.0-buildcache/install/intel/2021.10.0/fms-2024.02-m3n4l4t/lib/cmake/fms/fms-targets.cmake:69 (set_target_properties):
  The link interface of target "FMS::fms_r8" contains:

    OpenMP::OpenMP_C

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /home/ubuntu/spack-stack/manual-testing/spst-boost-gcc4intel/envs/ue-intel-2021.10.0-buildcache/install/intel/2021.10.0/fms-2024.02-m3n4l4t/lib/cmake/fms/fms-config.cmake:38 (include)
  fv3-jedi-lm/CMakeLists.txt:74 (find_package)
```
(This error message came from @climbfuji's attempt and running into the same issue).

It's not clear yet if this is an AWS instance issue, or it's an issue in general on all platforms.

**To Reproduce**

On an AWS Ubuntu instance:
1. build spack-stack from develop branches
2. build jedi-bundle

**Expected behavior**

The jedi-bundle build succeeds.

**System:**
What system(s) are you running the code on?
Ubuntu 22.04 AWS ECw instance
gcc, openmpi

**Additional context**
I found a temporary workaround, which is to edit the fms-targets.cmake file after the spack-stack build finishes, and remove the OpenMP_C target specs. In the `<spack-stack-env-dir>/install/fms-2024.02-<hash>/lib/cmake/fms/fms-targets.cmake` file, remove the OpenMP_C:
```
set_target_properties(FMS::fms_r4 PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include_r4"
  #INTERFACE_LINK_LIBRARIES "NetCDF::NetCDF_Fortran;MPI::MPI_Fortran;\$<LINK_ONLY:OpenMP::OpenMP_C>;\$<LINK_ONLY:OpenMP::OpenMP_Fortran>"
  INTERFACE_LINK_LIBRARIES "NetCDF::NetCDF_Fortran;MPI::MPI_Fortran;\$<LINK_ONLY:OpenMP::OpenMP_Fortran>"
)

# Create imported target FMS::fms_r8
add_library(FMS::fms_r8 STATIC IMPORTED)

set_target_properties(FMS::fms_r8 PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include_r8"
  #INTERFACE_LINK_LIBRARIES "NetCDF::NetCDF_Fortran;MPI::MPI_Fortran;\$<LINK_ONLY:OpenMP::OpenMP_C>;\$<LINK_ONLY:OpenMP::OpenMP_Fortran>"
  INTERFACE_LINK_LIBRARIES "NetCDF::NetCDF_Fortran;MPI::MPI_Fortran;\$<LINK_ONLY:OpenMP::OpenMP_Fortran>"
)
```
After making this change, the cmake error from running ecbuild on jedi-bundle went away, the jedi-bundle build succeeded and subsequent ctests passed for models using FMS. So there didn't seem to be an issue by doing this even though it seems rather dubious.

**Comment by srherbener** [comment]
After some digging I found that this change in the FMS cmake configuration added the OpenMP_C component to the exported targets: https://github.com/NOAA-GFDL/FMS/blob/f703b82972701a4e32a46d4d44a15f9fc2debb27/CMakeLists.txt#L413

The OpenMP_C target was deliberately put in to address a fault on MacOS: https://github.com/NOAA-GFDL/FMS/pull/1434

I'm pretty sure that the workaround I used is not appropriate, but it is curious that the FV3 code apparently didn't need the OpenMP_C component. 

Perhaps changing the find_package(OpenMP ...) calls in the model repos to ask for the "C" component is the better way to address this. I tried this in the fv3-jedi repo, it didn't work, and I probably didn't try enough places. 

The reason I'm putting all of this here in this issue, is that I need to hand this work off to someone as I will be out of the office both tomorrow and on Monday.

**Comment by climbfuji** [comment]
Yes, this would be my suggestion. Fix the find_package call in the jedi component that is missing it.

**Comment by eap** [comment]
I have tested a fix for this that works for spack-stack 1.8. I'll ensure they also pass CI tests in 1.7 and send for review.

**Comment by srherbener** [comment]
The fix for this is in: https://github.com/JCSDA-internal/fv3-jedi-linearmodel/pull/31 which has been merged. Closing as completed.
